phase2/src/circom_circuit.rs

```diff
@@ -363,3 +423,155 @@ pub fn circuit_from_json<E: Engine, R: Read>(reader: R) -> CircomCircuit::<E> {
 pub fn create_rng() -> Box<dyn Rng> {
     return Box::new(OsRng::new().unwrap())
 }
+
+
+
+
+fn circuit_from_r1cs_read_header<E: Engine, R:Read>(circuit : &mut CircomCircuit<E>, reader: &mut R) -> std::io::Result<()> {
+
+    let n8 = reader.read_u32::<LittleEndian>()?;
+    if n8 != 32 {
+        return Err(Error::new(ErrorKind::InvalidData, "Field size is not 256 bits."));
+    }
+
+    let mut q: [u64;4] = [0;4];
+    for i in 0..4 {
+        q[i] = reader.read_u64::<LittleEndian>()?;
+    }
+
+    if q != [ 
```

Comment:
What is this constant?

Reply:
This is the prime field of Fr in the bn254. 
This field is in the format because the format accepts any curv.

Reply:
Nice, thanks. Can you add a comment about it?